### DISCUSSION TITLE
What is "Serverless Function Execution" GB-Hrs

### DISCUSSION THREAD

**Discussion by meotimdihia** [vercel/vercel#5384] [Help] [answered]
Today, I received an email about the limit exceeded :

> your free account has used the following resources: 
> 
> • Serverless Execution: 167.7 GB-Hrs 
> • Build Hours: 2.77 hours 
> • Bandwidth: 3.26 GB 
> 

I read it here: https://vercel.com/docs/platform/fair-use-policy
I really don't understand what is "Serverless Function Execution" GB/hours? And how I can optimize it.

![msedge_2020-11-08_22-41-22](https://user-images.githubusercontent.com/300961/98469669-92f22d80-2213-11eb-8334-9fd044164161.png)

**Comment by meotimdihia** [comment]
I configured function memory. Hope I did it right.
https://vercel.com/docs/configuration#project/functions

**Comment by mcsdevv** [comment, marked as answer]
Hello @meotimdihia, "GB-Hrs" refers to the amount of time your Serverless Functions execute for when taking into account the memory available. This is calculated initially as "GB-s" and converted into "GB-Hrs" to make the figures much easier to reason about.

For example, if you have a function configured to use 1GB of memory and executes for 1 second, this would be billed at 1 GB-s. We'd need to have quite a few of these executions (3600) to reach a whole GB-Hr, however, it will always be calculated the same way.

As a further example, if you have a function configured to use 3GB of memory that executes for 1 second, this would be billed at 3GB-s. To reach 1 GB-Hr with this function, you'd need fewer executions, only 1200.

Therefore, the memory you use (defaulting to 1GB) has a direct impact on how fast you reach the limits you have referenced. Do note that these are somewhat contrived examples and the vast majority of Serverless Functions - as intended - will execute for a much shorter duration, with a minimal billed duration of 100ms.

Hope that helps!

**Reply by stackoverprof** [reply]
and how can i monitor all these execution statistic? where to see?

**Reply by Vampiro** [reply]
> and how can i monitor all these execution statistic? where to see?

You can see these in the Vercel Usage Dashboard: https://vercel.com/dashboard/usage
